### PR TITLE
Simple NDVI Timeseries example

### PR DESCRIPTION
**Pull request by Pratichhya**

### CODE REVIEW

File: openeo/NDVI_Timeseries.ipynb [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+     "output_type": "stream",
+     "text": [
+      "/home/pratixa/.local/lib/python3.6/site-packages/requests/__init__.py:104: RequestsDependencyWarning: urllib3 (1.26.15) or chardet (5.0.0)/charset_normalizer (2.0.12) doesn't match a supported version!\n",
+      "  RequestsDependencyWarning)\n"
+     ]
+    }
+   ],
+   "source": [
+    "import pandas as pd\n",
+    "import matplotlib.pyplot as pl\n",
```
Comment (soxofaan) [inline openeo/NDVI_Timeseries.ipynb:28]: you import matplolib twice here

File: openeo/NDVI_Timeseries.ipynb [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+    "fields = shapely.geometry.shape(json.loads(fields_geojson))\n",
+    "fields"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Like before, we load like a data cube and calculate the NDVI, but we now use a larger time window covering of a couple of months.\n",
```
Comment (soxofaan) [inline openeo/NDVI_Timeseries.ipynb:96]: > Like before, ...

what does this refer to?

File: openeo/NDVI_Timeseries.ipynb [hunk 3]
```diff
@@ -0,0 +1,427 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# defining the plotting function\n",
+    "\n",
+    "import matplotlib.pyplot as plt \n",
```
Comment (soxofaan) [inline openeo/NDVI_Timeseries.ipynb:260]: you import matplotlib here for a third time :)

File: openeo/NDVI_Timeseries.ipynb [hunk 4]
```diff
@@ -0,0 +1,427 @@
...
+    "DEFAULT_FIGSIZE = (5, 4)\n",
+    "\n",
+    "def plot_timeseries(filename):\n",
+    "    \"\"\"Helper to plot the timeseries directly from JSON file\"\"\"\n",
+    "    with open(filename) as f:\n",
+    "        ts = timeseries_json_to_pandas(json.load(f)).dropna()\n",
```
Comment (soxofaan) [inline openeo/NDVI_Timeseries.ipynb:266]: you are doing all this `timeseries_json_to_pandas(json.load(f)).dropna()` stuff again here. Can't you just reuse what you did a couple of cells before?

File: openeo/NDVI_Timeseries.ipynb [hunk 5]
```diff
@@ -0,0 +1,427 @@
...
+    "The result above is a good start, but needs some work: there are quite some outliers and zero's that don't look right.\n",
+    "The reason for this are cloudy observations that cause unusable NDVI values.\n",
+    "We will add some cloud masking to generate a cleaner time series."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First, create a simple gaussian kernel:"
```
Comment (soxofaan) [inline openeo/NDVI_Timeseries.ipynb:309]: The explanation stops quite abruptly here. It is not clear why you need a gaussian kernel and what happens with it

File: openeo/NDVI_Timeseries_UDF_Smoothing.ipynb [hunk 6]
```diff
@@ -0,0 +1,296 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Timeseries Smoothing\n",
+    "\n",
+    "This notebook is a supplementary notebook to NDVI_Timeseries.ipynb, where we aim to improve the plot's smoothness by using a kernel as an UDF. The kernel will help to reduce noise and fluctuations in the data, resulting in a smoother and more accurate representation of the NDVI timeseries."
```
Comment (soxofaan) [inline openeo/NDVI_Timeseries_UDF_Smoothing.ipynb:9]: Why starting a whole new notebook with a lot of duplication?
Isn't it cleaner to just continue under the other one with a new section/chapter?

File: openeo/NDVI_Timeseries_UDF_Smoothing.ipynb [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+    "smoothed_timeseries = smoothed_ndvi.aggregate_spatial(geometries=fields, reducer=\"mean\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "smoothed_timeseries.download(\"output/timeseries-smoothed.json\", format=\"JSON\")"
```
Comment (soxofaan) [inline openeo/NDVI_Timeseries_UDF_Smoothing.ipynb:244]: I would drop the unnecessary `format=JSON`

File: openeo/NDVI_Timeseries.ipynb (hunk 1 above)
Comment (Pratichhya) [reply openeo/NDVI_Timeseries.ipynb:28]: updated

File: openeo/NDVI_Timeseries.ipynb (hunk 2 above)
Comment (Pratichhya) [reply openeo/NDVI_Timeseries.ipynb:96]: updated

File: openeo/NDVI_Timeseries.ipynb [hunk 8]
```diff
@@ -0,0 +1,427 @@
...
+    "Because `DataCube.aggregate_spatial()` returns a timeseries (instead of raster data), we download it in JSON format."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "timeseries.download(\"output/timeseries.json\", format=\"JSON\")"
```
[the remark this answers is not in the document]
Comment (Pratichhya) [reply openeo/NDVI_Timeseries.ipynb:148]: updated

File: openeo/NDVI_Timeseries.ipynb (hunk 3 above)
Comment (Pratichhya) [reply openeo/NDVI_Timeseries.ipynb:260]: corrected

File: openeo/NDVI_Timeseries.ipynb (hunk 4 above)
Comment (Pratichhya) [reply openeo/NDVI_Timeseries.ipynb:266]: removed unwanted section

File: openeo/NDVI_Timeseries.ipynb (hunk 5 above)
Comment (Pratichhya) [reply openeo/NDVI_Timeseries.ipynb:309]: updated

File: openeo/NDVI_Timeseries_UDF_Smoothing.ipynb (hunk 7 above)
Comment (Pratichhya) [reply openeo/NDVI_Timeseries_UDF_Smoothing.ipynb:244]: updated

File: openeo/NDVI_Timeseries_UDF_Smoothing.ipynb (hunk 6 above)
Comment (Pratichhya) [reply openeo/NDVI_Timeseries_UDF_Smoothing.ipynb]: combined as one.

**Review by soxofaan — APPROVED** [review]

**Pull request merged**